### PR TITLE
SUS-1257: don't allow blocked users to issue chat bans

### PR DESCRIPTION
**Pull request by mszabo-wikia**
also changed error messages to use user language instead of content language, since they're only shown to the user

ticket: https://wikia-inc.atlassian.net/browse/SUS-1257

### CODE REVIEW

File: extensions/wikia/Chat2/Chat.class.php [hunk 1]
```diff
@@ -87,7 +87,9 @@ public static function getSessionKey() {
 	 *
 	 * @return true|string Returns true on success, returns an error message as a string on failure.
 	 */
-	public static function banUser( $subjectUserName, User $adminUser, $time, $reason ) {
+	public static function banUser( string $subjectUserName, User $adminUser, int $time, string $reason ) {
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/Chat.class.php:90]: MAJOR Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 13 issues
* CRITICAL 1 critical
* MAJOR 3 major
* MINOR 6 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [Chat.class.php#L164](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L164): Move arguments "$dir" after arguments without default value [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. MAJOR [Chat.class.php#L13](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L13): Class "Chat" has 25 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [Chat.class.php#L470](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L470): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MINOR [Chat.class.php#L64](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L64): Define a constant instead of duplicating this literal ": Method called" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [Chat.class.php#L218](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L218): Immediately return this expression instead of assigning it to the temporary variable "$result". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L243](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L243): Immediately return this expression instead of assigning it to the temporary variable "$errorMsg". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L258](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L258): Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. INFO [Chat.class.php#L18](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L18): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L19](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L19): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L353](https://github.com/Wikia/app/blob/d178bfaeb60402aff450e51879afb6ca0fd51288/extensions/wikia/Chat2/Chat.class.php#L353): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues
* MAJOR 3 major
* MINOR 6 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Chat.class.php#L13](https://github.com/Wikia/app/blob/ca62b1b8aeda87afba111f9bb5acc8cf5fadaaa1/extensions/wikia/Chat2/Chat.class.php#L13): Class "Chat" has 25 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [Chat.class.php#L470](https://github.com/Wikia/app/blob/ca62b1b8aeda87afba111f9bb5acc8cf5fadaaa1/extensions/wikia/Chat2/Chat.class.php#L470): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MINOR [Chat.class.php#L64](https://github.com/Wikia/app/blob/ca62b1b8aeda87afba111f9bb5acc8cf5fadaaa1/extensions/wikia/Chat2/Chat.class.php#L64): Define a constant instead of duplicating this literal ": Method called" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [Chat.class.php#L218](https://github.com/Wikia/app/blob/ca62b1b8aeda87afba111f9bb5acc8cf5fadaaa1/extensions/wikia/Chat2/Chat.class.php#L218): Immediately return this expression instead of assigning it to the temporary variable "$result". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L243](https://github.com/Wikia/app/blob/ca62b1b8aeda87afba111f9bb5acc8cf5fadaaa1/extensions/wikia/Chat2/Chat.class.php#L243): Immediately return this expression instead of assigning it to the temporary variable "$errorMsg". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L258](https://github.com/Wikia/app/blob/ca62b1b8aeda87afba111f9bb5acc8cf5fadaaa1/extensions/wikia/Chat2/Chat.class.php#L258): Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. INFO [Chat.class.php#L18](https://github.com/Wikia/app/blob/ca62b1b8aeda87afba111f9bb5acc8cf5fadaaa1/extensions/wikia/Chat2/Chat.class.php#L18): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L19](https://github.com/Wikia/app/blob/ca62b1b8aeda87afba111f9bb5acc8cf5fadaaa1/extensions/wikia/Chat2/Chat.class.php#L19): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L353](https://github.com/Wikia/app/blob/ca62b1b8aeda87afba111f9bb5acc8cf5fadaaa1/extensions/wikia/Chat2/Chat.class.php#L353): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 12 issues
* MAJOR 3 major
* MINOR 6 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Chat.class.php#L13](https://github.com/Wikia/app/blob/5c35e306ef3ebeaefa298ed74ce53df6b7c0bb99/extensions/wikia/Chat2/Chat.class.php#L13): Class "Chat" has 25 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [Chat.class.php#L470](https://github.com/Wikia/app/blob/5c35e306ef3ebeaefa298ed74ce53df6b7c0bb99/extensions/wikia/Chat2/Chat.class.php#L470): Reduce the number of returns of this function 5, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MINOR [Chat.class.php#L64](https://github.com/Wikia/app/blob/5c35e306ef3ebeaefa298ed74ce53df6b7c0bb99/extensions/wikia/Chat2/Chat.class.php#L64): Define a constant instead of duplicating this literal ": Method called" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [Chat.class.php#L218](https://github.com/Wikia/app/blob/5c35e306ef3ebeaefa298ed74ce53df6b7c0bb99/extensions/wikia/Chat2/Chat.class.php#L218): Immediately return this expression instead of assigning it to the temporary variable "$result". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L243](https://github.com/Wikia/app/blob/5c35e306ef3ebeaefa298ed74ce53df6b7c0bb99/extensions/wikia/Chat2/Chat.class.php#L243): Immediately return this expression instead of assigning it to the temporary variable "$errorMsg". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L258](https://github.com/Wikia/app/blob/5c35e306ef3ebeaefa298ed74ce53df6b7c0bb99/extensions/wikia/Chat2/Chat.class.php#L258): Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. INFO [Chat.class.php#L18](https://github.com/Wikia/app/blob/5c35e306ef3ebeaefa298ed74ce53df6b7c0bb99/extensions/wikia/Chat2/Chat.class.php#L18): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L19](https://github.com/Wikia/app/blob/5c35e306ef3ebeaefa298ed74ce53df6b7c0bb99/extensions/wikia/Chat2/Chat.class.php#L19): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L353](https://github.com/Wikia/app/blob/5c35e306ef3ebeaefa298ed74ce53df6b7c0bb99/extensions/wikia/Chat2/Chat.class.php#L353): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* MAJOR 2 major
* MINOR 6 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Chat.class.php#L13](https://github.com/Wikia/app/blob/cb602368dde6c011f9c70dadfb608ad91772c665/extensions/wikia/Chat2/Chat.class.php#L13): Class "Chat" has 25 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MINOR [Chat.class.php#L64](https://github.com/Wikia/app/blob/cb602368dde6c011f9c70dadfb608ad91772c665/extensions/wikia/Chat2/Chat.class.php#L64): Define a constant instead of duplicating this literal ": Method called" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [Chat.class.php#L220](https://github.com/Wikia/app/blob/cb602368dde6c011f9c70dadfb608ad91772c665/extensions/wikia/Chat2/Chat.class.php#L220): Immediately return this expression instead of assigning it to the temporary variable "$result". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L245](https://github.com/Wikia/app/blob/cb602368dde6c011f9c70dadfb608ad91772c665/extensions/wikia/Chat2/Chat.class.php#L245): Immediately return this expression instead of assigning it to the temporary variable "$errorMsg". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L260](https://github.com/Wikia/app/blob/cb602368dde6c011f9c70dadfb608ad91772c665/extensions/wikia/Chat2/Chat.class.php#L260): Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. INFO [Chat.class.php#L18](https://github.com/Wikia/app/blob/cb602368dde6c011f9c70dadfb608ad91772c665/extensions/wikia/Chat2/Chat.class.php#L18): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L19](https://github.com/Wikia/app/blob/cb602368dde6c011f9c70dadfb608ad91772c665/extensions/wikia/Chat2/Chat.class.php#L19): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L355](https://github.com/Wikia/app/blob/cb602368dde6c011f9c70dadfb608ad91772c665/extensions/wikia/Chat2/Chat.class.php#L355): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* MAJOR 2 major
* MINOR 6 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Chat.class.php#L13](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L13): Class "Chat" has 25 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MINOR [Chat.class.php#L64](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L64): Define a constant instead of duplicating this literal ": Method called" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [Chat.class.php#L220](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L220): Immediately return this expression instead of assigning it to the temporary variable "$result". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L245](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L245): Immediately return this expression instead of assigning it to the temporary variable "$errorMsg". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)
1. MINOR [Chat.class.php#L260](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L260): Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. INFO [Chat.class.php#L18](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L18): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L19](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L19): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. INFO [Chat.class.php#L355](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L355): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

**Comment by ghost** [comment]
Per QA, this one needs more work. This is waiting for SUS to take another round with Chat.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 21 issues
* ![MAJOR][MAJOR] 15 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Chat.class.php#L13](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L13): Class "Chat" has 25 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [Chat.class.php#L67](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L67): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [Chat.class.php#L70](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L70): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [Chat.class.php#L72](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L72): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [Chat.class.php#L347](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L347): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [Chat.class.php#L351](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L351): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [Chat.class.php#L417](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L417): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [Chat.class.php#L418](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L418): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [Chat.class.php#L425](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L425): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [Chat.class.php#L427](https://github.com/Wikia/app/blob/87e144b9037aa994def0b62e42b489f9dab16774/extensions/wikia/Chat2/Chat.class.php#L427): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**